### PR TITLE
add ignored keyword filtering

### PR DESCRIPTION
**Pull request by erquhart**
It compiles, but I didn't create and test binaries. Hopefully a go dev can approve without running it.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -50,6 +52,15 @@ func init() {
 func decode(conn *twitterstream.Connection) {
 	for {
 		if tweet, err := conn.Next(); err == nil {
+
+			// Filter out ignored keywords
+			for _, keyword := range strings.Split(config.KeywordsIgnored, ",") {
+				if strings.Contains(tweet.Text, keyword) {
+					log.Printf("Tweet contained ignored keyword: %s", keyword)
+					return
```
Comment (calavera) [inline main.go]: don't return here, use `break`. I'm pretty sure we want to keep looping.

**Review by calavera — CHANGES_REQUESTED** [review]

File: config.example.json [hunk 2]
```diff
@@ -5,5 +5,6 @@
   "accessSecret": "TWITTER ACCESS SECRET",
   "slackWebhook": "SLACK INCOMING WEBHOOK",
   "slackChannel": "OPTIONAL SLACK CHANNEL TO SEND TO",
-  "keywords": "keywords,your want,to track"
+  "keywords": "keywords,your want,to track",
+  "keywordsIgnored": "keywords,you do not want,to track"
```
Comment (calavera) [inline config.example.json]: json supports lists of strings, why don't we use them rather than splitting this string every time?

Comment (erquhart) [reply config.example.json]: Yeah that makes sense.

File: main.go (hunk 1 above)
Comment (erquhart) [reply main.go]: This function decodes a single tweet, so we want to stop if that tweet has an ignored keyword, right?

Comment (calavera) [reply main.go]: I meant `continue`, not `break`, sorry for the confusion

File: config.example.json (hunk 2 above)
Comment (verythorough) [reply config.example.json]: This is silly and small, but the typo in line 8's "your want" actually did make it harder for me to understand until I read line 9's "you do not want"

File: main.go (hunk 1 above)
Comment (calavera) [reply main.go]: you want to continue the outer loop, this function decodes all tweets coming from the connection. `Next` is a live streaming feed waiting for messages.

File: config.example.json (hunk 2 above)
Comment (erquhart) [reply config.example.json]: I'll correct that.

@calavera I remembered why I did this - we're not touching the keywords string (it's consumed by Twitter's api), and I didn't want to have two different signatures for `keywords` and `keywordsIgnored`. The related code is almost negligible in size.

Comment (calavera) [reply config.example.json]: ok

**Comment by erquhart** [comment]
Adding this comment from mobile - this probably doesn't allow the config to not have the keywordsIgnore field. I'll update in the am.

**Comment by erquhart** [comment]
@calavera tested live, good to go.

**Review by calavera — APPROVED** [review]

**Pull request merged**